keel-intents/src/main/kotlin/com/netflix/spinnaker/keel/intents/LoadBalancerIntent.kt

```diff
@@ -0,0 +1,109 @@
+package com.netflix.spinnaker.keel.intents
+
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.fasterxml.jackson.annotation.JsonTypeInfo.As
+import com.fasterxml.jackson.annotation.JsonTypeInfo.As.PROPERTY
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME
+import com.fasterxml.jackson.annotation.JsonTypeName
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.fasterxml.jackson.databind.annotation.JsonSerialize
+import com.github.jonpeterson.jackson.module.versioning.JsonVersionedModel
+import com.netflix.spinnaker.keel.ApplicationAwareIntentSpec
+import com.netflix.spinnaker.keel.Intent
+import com.netflix.spinnaker.keel.intents.AvailabilityZoneConfig.Automatic
+import com.netflix.spinnaker.keel.intents.HealthEndpoint.*
+import com.netflix.spinnaker.keel.intents.jackson.AvailabilityZoneConfigDeserializer
+import com.netflix.spinnaker.keel.intents.jackson.AvailabilityZoneConfigSerializer
+import com.netflix.spinnaker.keel.model.Listener
+import com.netflix.spinnaker.keel.model.Protocol
+import com.netflix.spinnaker.keel.model.Protocol.*
+import com.netflix.spinnaker.keel.model.Scheme
+
+private const val KIND = "LoadBalancer"
+private const val CURRENT_SCHEMA = "1"
+
+@JsonTypeName(KIND)
+@JsonVersionedModel(currentVersion = CURRENT_SCHEMA, propertyName = SCHEMA_PROPERTY)
+class LoadBalancerIntent(spec: LoadBalancerSpec) : Intent<LoadBalancerSpec>(
+  kind = KIND,
+  schema = CURRENT_SCHEMA,
+  spec = spec
+) {
+  override val id = "$KIND:${spec.cloudProvider}:${spec.accountName}:${spec.name}"
+}
+
+@JsonTypeInfo(use = NAME, include = PROPERTY, property = KIND_PROPERTY)
+abstract class LoadBalancerSpec : ApplicationAwareIntentSpec() {
+  abstract val name: String
+  abstract val cloudProvider: String
+  abstract val accountName: String
+  abstract val region: String
```

Comment:
One thing I did with security groups is changed `region` -> `regions` then I flatMap on the regions value. I think this could be valuable to reduce repetition. WDYT, or should we have jsonnet handle this?

Reply:
I saw that and was assuming that was what you were doing. Doesn't that mean the conversion from keel -> cloud driver is 1..n?

Reply:
I was assuming the conversions should be symmetrical but maybe that's not necessarily the case. When do we want to convert from CloudDriver -> Keel model?